python/pyspark/errors/utils.py

```diff
@@ -0,0 +1,133 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import json
+import re
+from typing import Dict
+
+
+ERROR_CLASSES_JSON = "error-classes.json"
+ERROR_CLASSES_PATH = os.path.join(os.path.dirname(__file__), ERROR_CLASSES_JSON)
+
+
+class ErrorClassesJsonReader:
+    """
+    A reader to load error information from JSON file.
+    """
+
+    def __init__(self, json_file_path: str):
+        try:
+            self.error_info_map = json.load(open(json_file_path))
+        except NotADirectoryError:
```

Comment:
This is too flaky to rely on `NotADirectoryError` system exception. The directory might exist from users' file system as an example.